Std/Data/Bool.lean

```diff
@@ -72,8 +76,13 @@ theorem and_xor_distrib_right : ∀ (x y z : Bool), (xor x y && z) = xor (x && z
 /-- De Morgan's law for boolean and -/
 theorem not_and : ∀ (x y : Bool), (!(x && y)) = (!x || !y) := by decide
 
+@[deprecated and_eq_true]
 theorem and_eq_true_iff : ∀ (x y : Bool), (x && y) = true ↔ x = true ∧ y = true := by decide
```

Comment:
Here and elsewhere, the `Iff` versions are useful since they gives access to the two implications via `.mp` and `.mpr`. This doesn't work with the `Eq` version. 

That said, they would be more useful if the variables were marked implicit, as they would with the two implications.

Reply:
That helps me understand the conventions here.  Thanks

Reply:
This is on pause for a bit while I work through a comprehensive look at `Bool`/`Prop`, but I noticed that `Eq` also give access to the implications via `Eq.mp` and `Eq.mpr`.

I think I want to systematically use one (probably `Eq`), but that's not settled just yet.